http-clients/java-nio-client/src/test/java/software/amazon/awssdk/http/nio/java/internal/ListToByteBufferProcessorTest.java

```diff
@@ -0,0 +1,135 @@
+package software.amazon.awssdk.http.nio.java.internal;
+
+
+import static junit.framework.TestCase.assertEquals;
+import io.reactivex.FlowableSubscriber;
+import io.reactivex.processors.PublishProcessor;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Flow;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.reactivestreams.FlowAdapters;
+import org.reactivestreams.Subscription;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ListToByteBufferProcessorTest {
```

Comment:
We should be testing the publisher. Ideally, this is what the test should do:

- Create a `Publisher<List<ByteBuffer>>` with at least two Lists. Use RxJava for this. Something like:

```java
bufferListPublisher = Flowable.just(list1, list2);
```
- Subscribe `listToByteBufferProcessor` to the `Publisher` created in the above step

```java
bufferListPublisher.subscribe(listToByteBufferProcessor);
```
- Use RxJava to verify the contents of the publisher returned by `listToByteBufferProcessor`

```java
Flowable.fromPublisher(listToByteBufferProcessor.getPublisherToSdk()).toList()
```